File: src/main/scala/org/unipi/matrixnorm/FakeInputSplit.scala
package org.unipi.matrixnorm

import org.apache.hadoop.mapreduce.InputSplit
import org.apache.hadoop.io.Writable
import java.io._

class FakeInputSplit extends InputSplit with Writable {

  @throws[IOException]
  def readFields(var1: DataInput): Unit = {

  }

  @throws[IOException]
  def write(var1: DataOutput): Unit = {

  }

  @throws[IOException]
  @throws[InterruptedException]
  def getLength: Long = {
    0L
  }

  @throws[IOException]
  @throws[InterruptedException]
  def getLocations = new Array[String](0)

}

File: src/main/java/org/unipi/matrixgen/FakeInputSplit.java
package org.unipi.matrixgen;

import org.apache.hadoop.io.Writable;
import org.apache.hadoop.mapreduce.InputSplit;

import java.io.DataInput;
import java.io.DataOutput;
import java.io.IOException;

public class FakeInputSplit extends InputSplit implements Writable {

    public void readFields(DataInput arg0) throws IOException {
    }

    public void write(DataOutput arg0) throws IOException {
    }

    public long getLength() throws IOException, InterruptedException {
        return 0;
    }

    public String[] getLocations() throws IOException,
            InterruptedException {
        return new String[0];
    }
}
